pallets/identity/src/lib.rs

```diff
@@ -255,6 +261,31 @@ decl_module! {
             use polymesh_primitives::{identity_claim::IdentityClaimOld, migrate::migrate_map};
             migrate_map::<IdentityClaimOld>(b"Identity", b"Claims");
 
+            // Covert old scopes to new scopes
+            use frame_support::migration::{StorageIterator, put_storage_value};
+            let old_map = StorageIterator::<IdentityClaim>::new(b"Identity", b"Claims")
+                .drain().
+                collect::<Vec<(Vec<u8>, IdentityClaim)>>();
+
+            for (raw_key, value) in old_map.iter() {
+                let mut unhashed_key = Blake2_128Concat::reverse(&raw_key);
+                if let Ok(k1) = Claim1stKey::decode(&mut unhashed_key) {
```

Comment:
Ostensibly, we should `.unwrap()` here cause there's a serious problem if decoding fails due to e.g., wrong type, so crashing loudly is I think preferable to silently dropping data. (Applies to 2nd `.decode(...)` too of course.)

Reply:
@maxsam4 thoughts?

Reply:
In general, we should never be crashing loudly because if that happens, the blockchain will be stuck and we won't be able to fix it.

We should always let the blockchain continue so that we can submit a new upgrade that fixes past mistakes.